SYCL/KernelAndProgram/level-zero-link-flags.cpp

```diff
@@ -0,0 +1,54 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER SYCL_PROGRAM_LINK_OPTIONS=-foo %t.out
+// REQUIRES: level_zero
+//
+// This test is disabled because the runtime does not currently pass linker
+// flags from SYCL_PROGRAM_LINK_OPTIONS when the program calls "sycl::link()".
+// This seems like a bug since the runtime passes linker flags from that
+// environment variable in other cases when it links device code.
```

Comment:
@smaslov-intel: Do you agree that the behavior I describe above is a bug?  If so, I'll create a bug report.

Reply:
I am not sure, the SYCL_PROGRAM_LINK_OPTIONS doc says "String of valid OpenCL link options", so it sounds like OpenCL specific. Also, given that Level-Zero doesn't support any linking options, why would we have an option to pass any?

Reply:
The handling for `SYCL_PROGRAM_LINK_OPTIONS` is done in the generic part of the DPC++ runtime.  Currently, it only uses this envvar when a module is compiled in order to submit it to a device.  In that case, the runtime passes the `SYCL_PROGRAM_LINK_OPTIONS` to `piProgramLink` regardless of the backend.  Therefore, I think the intent of this envvar is that it can be used to pass options to the linker for *any* backend, not just for OpenCL.

The handling for `sycl::link()` is also in the generic part of the library.  For some reason, that code does not use options from `SYCL_PROGRAM_LINK_OPTIONS` for any backend.

Reply:
I am OK making SYCL_PROGRAM_LINK_OPTIONS to be backend-agnostic (need to fix doc then).

Reply:
I created a bug report, which also captures the doc issue.

Reply:
I got feedback that I should pass the linker options via [`-Xsycl-target-linker`](https://intel.github.io/llvm-docs/UsersManual.html#target-toolchain-options) instead of using `SYCL_PROGRAM_LINK_OPTIONS `.  I modified the test to use that option, but it still fails.  Therefore, this test is still disabled.  I entered a new bug report to capture this issue.

Reply:
@smaslov-intel said:

> Why is the test still disabled?

I assume you are asking about the "level-zero-link-flags.cpp" test?  If so, my comment above is the answer.